internal/api/grpc/server/middleware/auth_interceptor.go

```diff
@@ -26,7 +26,7 @@ func authorize(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
 	}
 
 	authCtx, span := tracing.NewServerInterceptorSpan(ctx)
-	defer func() { span.EndWithError(err) }()
+	defer span.EndWithError(err)
```

Comment:
Doesn't this evaluate the `err` variable prematurely and then `err` will always be reported as `nil` even if it's value changes later.

> Each time a "defer" statement executes, the function value and parameters to the call are [evaluated as usual](https://go.dev/ref/spec#Calls) and saved anew but the actual function is not invoked

https://go.dev/ref/spec#Defer_statements
https://stackoverflow.com/questions/24720097/golang-defer-behavior